### PR TITLE
gui.auto_commit: allow callback specified other than commit()

### PR DESCRIPTION
**Pull request by kernc**
Not tested beyond it's still working for widgets using it as currently and my own use case.

### CODE REVIEW

**Comment by janezd** [comment]
Is there another rationale for this than allowing a different name?

File: Orange/widgets/gui.py [hunk 1]
```diff
@@ -2239,7 +2239,7 @@ def paintEvent(self, ev):
 
 
 def auto_commit(widget, master, value, label, auto_label=None, box=True,
-                checkbox_label=None, orientation=None, **misc):
+                checkbox_label=None, orientation=None, callback='commit', **misc):
```
Comment (janezd) [inline Orange/widgets/gui.py]: I would prefer `callback=None` as the default argument. Furthermore, I don't like passing strings like this; we have to do it for attributes, while for methods we can pass the reference, e.g. `callback=self.apply`. The method's name is then retrieved through `callback.__name__`.

File: Orange/widgets/gui.py [hunk 2]
```diff
@@ -2293,12 +2295,12 @@ def commit():
     def do_commit():
         nonlocal dirty
         QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
-        master.unconditional_commit()
+        getattr(master, 'unconditional_' + callback)()
```
Comment (janezd) [inline Orange/widgets/gui.py]: If the method is given a reference instead of string, this line can be replaced by `callback()`.

**Comment by kernc** [comment]
It's for another button that does something else than "commit." Say I have two such auto-able buttons on a widget.

**Comment by janezd** [comment]
I thought about this, but didn't want to over-design. If you have a use case with two buttons, I agree. If you just want to rename, I hesitatingly agree. :)

**Comment by kernc** [comment]
Renaming wouldn't warrant a pull request. I just cringe and try to live with it.

**Pull request merged**